From 20ab712989e4c2c3afe32b18a807cb4f02a9bf22 Mon Sep 17 00:00:00 2001
From: Hung Tran <hung.tran.jy@renesas.com>
Date: Thu, 5 Oct 2017 20:47:10 +0900
Subject: [PATCH 162/628] rcar-du: Work around to enable iwg20m LCD display

On iwg20m, when there is no Daughterboard, DU cannot
initialize encoder for HDMI.
However, the return code make kernel retry again
many times, instead of continue initialize next output.
This makes the LCD display not usable.

This patch works around this by changing the return
code, thus prevent retry.

Signed-off-by: Hung Tran <hung.tran.jy@renesas.com>
Signed-off-by: Nguyen Van Linh [FGA.BU13] <LinhNV25@fsoft.com.vn>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 drivers/gpu/drm/rcar-du/rcar_du_encoder.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_encoder.c b/drivers/gpu/drm/rcar-du/rcar_du_encoder.c
index 4e939e4..ff17176 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_encoder.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_encoder.c
@@ -201,5 +201,11 @@ done:
 		devm_kfree(rcdu->dev, renc);
 	}
 
+	if (ret == -EPROBE_DEFER)
+		/* Return -EPROBE_DEFER will make kernel trying to initialize
+		 * current output again. But in certain case it causes looping
+		 * and kernel cannot init another output. Disable temporarily */
+		ret = -1;
+
 	return ret;
 }
-- 
2.7.4

